cocos2d/core/camera/CCCamera.js

```diff
@@ -470,14 +470,21 @@ let Camera = cc.Class({
          * @static
          */
         findCamera (node) {
+            let camera=null;
             for (let i = 0, l = _cameras.length; i < l; i++) {
-                let camera = _cameras[i];
+                 camera = _cameras[i];
                 if (camera.containsNode(node)) {
                     return camera;
                 }
             }
-
-            return null;
+                if(!camera){
+                    if(_cameras.length>0) camera=_cameras[0]; 
+                    /// return the first camera instead of returning null. EditBox raises error if no camera found , 
+                        ///this may also prevent some other bugs in some other cases. Generally, gamedev would not want to check if the camera is null.. 
+                    /// return the first camera to precalculate even not active objects, thus preventing exceptions. ...
```

Comment:
I am assuming the precalulation is meaningless and will eclipse some real bug. Indeed EditBox raises error, and some other codes in the engine also forget to check it.
@PPpro  **We should improve the api description, add some tips for devs, and search all findCamera usage to ensure no error will result event if no active camera in the scene**

Reply:
in fact , the issue appears without developer call. The exact situation is :

There are 2 nodes in stack , same size and same position. 

Player wants to switch those nodes as fast as possible, so , instead of activating/deactivating them, we use camera culling and blockInputEvents component. An other  advantage of doing this is web based components are not refreshed.

When user loads the scene , web based components are culled by camera , so when engine code tries to calculate the position of that component, it calls position calculating function from the engine , which tries to call the null camera's sub functions. 

The best solution will of course be to not calculate culled web based components' position by default . My solution is just a workaround , but the bug is still there :)

Reply:
> for example webviev reloads and loses content, when activated/deactivated

So you just use camera to simulate the blink effect? It's so weird.



> engine tries to calculate webviews' position, the node containing web based components are culled by camera , so when engine code tries to calculate the position of that component, it calls position calculating function from the engine , which tries to call the null camera's sub functions, throwing an exception.

It's a bug of the engine, we will fix it to ensure no error will result.